app/src/main/java/edu/uwstout/p2pchat/NickNameFragment.java

```diff
@@ -0,0 +1,272 @@
+package edu.uwstout.p2pchat;
+
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.view.Display;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import java.util.ArrayList;
+
+
+
+/**
+ * NickNameFragment is the fragment responsible for showing the previously contacted
+ * nicknames.
+ * A simple {@link Fragment} subclass.
+ */
+public class NickNameFragment extends Fragment
+{
+    /**
+     * The listview that had been initilized in the xml file.
+     */
+    private ListView view;
+
+    /**
+     * context that the nickname fragment will be using.
+     */
+    private Context classContext;
+
+    /**
+     * Arraylist of names.
+     */
+    private ArrayList<String> names = new ArrayList<>();
+
+
+    private ArrayAdapter adapter;
+    /**
+     * Default constructor. Nothing will be used here.
+     */
+    public NickNameFragment()
+    {
+        //Nothing will be done here.
+    }
+
+
+    /**
+     *
+     * @param inflater inflator that came with with this method.
+     * @param container a container that came with this method.
+     * @param savedInstanceState just a bundle that came with this method.
+     * @return view for the nickname.
+     */
+    @Override
+    public View onCreateView(final LayoutInflater inflater, final ViewGroup container,
+            final Bundle savedInstanceState)
+    {
+        setNames();
+
+        //Inflate a view.
+        View view = inflater.inflate(R.layout.fragment_nick_name, container, false);
+
+        //Get the context from the view.
+        classContext = view.getContext();
+
+        //Put out on the screen that this is lauching.
+        Log.i("Launching", "Nickname view had launched");
```

Comment:
tag is not appropriate, log tag should be the name of the class and should be a class level static constant.